### PR TITLE
Add azure-pipelines.yml

### PR DESCRIPTION
**Pull request by tjprescott**
First attempt to get azdev switched over to using ADO.

### CODE REVIEW

File: tox.ini [hunk 1]
```diff
@@ -1,8 +1,5 @@
 [tox]
-envlist = 
-    py27
-    py35
-    py36
+envlist = ALL
```
Comment (marstr) [inline tox.ini]: I'd keep this as an enumerated list of the Python versions that we declare support for. That way folks using `tox` locally know which python interpreters they need installed on their machines.

File: azure-pipelines.yml [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+    include:
+    - '*'
+jobs:
+- job: Tox27
+  displayName: 'Tox: Python 2.7'
+  condition: succeeded()
+  pool:
+    vmImage: 'ubuntu-16.04'
+  steps:
+  - UsePythonVersion@0
```
Comment (marstr) [inline azure-pipelines.yml]: I believe you're missing some syntax here. It should be:
``` yaml
- task: UsePythonVersion@0
```

File: azure-pipelines.yml [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+    displayName: 'Use Python 2.7'
+    inputs:
+      versionSpec: 2.7
+  - task: Bash@3
+    displayName: 'Run Tox'
+    inputs:
+      targetType: 'filePath'
+      filePath: scripts/ci/run_tox.sh
```
Comment (marstr) [inline azure-pipelines.yml]: This is where I'd override the TOXENV environment variable to inform tox it should only run in one spot. That would make this section:

``` yaml
- task: Bash@3
  displayName: 'Run Tox'
  env:
    TOXENV: py27
  inputs:
    targetType: 'filePath'
    filePath: scripts/ci/run_tox.sh
```
The other Python versions should follow suit as well, of course.

I'm sure you've already seen it, but the documentation I'm working from can be found here: [tox.readthedocs.io#selecting-one-or-more-environments](https://tox.readthedocs.io/en/latest/example/general.html#selecting-one-or-more-environments-to-run-tests-against)

File: tox.ini (hunk 1 above)
Comment (tjprescott) [reply tox.ini]: Ah, good point.

**Pull request merged**